src/main/java/edu/kit/provideq/toolbox/tsp/solvers/QuboTspSolver.java

```diff
@@ -0,0 +1,189 @@
+  private static final SubRoutineDefinition<String, String> QUBO_SUBROUTINE =
+      new SubRoutineDefinition<>(QuboConfiguration.QUBO, "How should the QUBO be solved?");
+  private final ApplicationContext context;
+  private final String binaryDir;
+  private final String binaryName;
+  private ResourceProvider resourceProvider;
+
+
+  @Autowired
+  public QuboTspSolver(
+      //"vrp" value is correct because this uses the VRP framework from Lucas Bergers thesis
+      @Value("${custom.berger-vrp.directory}") String binaryDir,
+      @Value("${custom.berger-vrp.solver}") String binaryName,
+      ApplicationContext context) {
+    this.binaryName = binaryName;
+    this.binaryDir = binaryDir;
+    this.context = context;
+  }
+
+  public static boolean checkVehicleCapacity(String tsp) {
+    int capacity = 0;
+    int totalDemand = 0;
+    boolean demandSection = false;
+
+    Pattern capacityPattern = Pattern.compile("CAPACITY\\s*:\\s*(\\d+)");
+    Pattern demandPattern = Pattern.compile("^\\d+\\s*(\\d+)");
+
+    for (String line : tsp.split("\n")) {
+      Matcher capacityMatcher = capacityPattern.matcher(line);
+      if (capacityMatcher.find()) {
+        capacity = Integer.parseInt(capacityMatcher.group(1));
+      }
+      if (line.startsWith("DEMAND_SECTION")) {
+        demandSection = true;
+        continue;
+      }
+      if (line.startsWith("EOF")) {
+        demandSection = false;
+      }
+      if (demandSection) {
+        Matcher demandMatcher = demandPattern.matcher(line);
+        if (demandMatcher.find()) {
+          totalDemand += Integer.parseInt(demandMatcher.group(1));
+        }
+      }
+    }
+
+    return totalDemand <= capacity;
+  }
+
+  @Autowired
+  public void setResourceProvider(ResourceProvider resourceProvider) {
+    this.resourceProvider = resourceProvider;
+  }
+
+  @Override
+  public String getName() {
+    return "TSP to QUBO Transformation";
+  }
+
+  @Override
+  public List<SubRoutineDefinition<?, ?>> getSubRoutines() {
+    return List.of(QUBO_SUBROUTINE);
+  }
+
+  private String getProblemDirectory(Solution<String> solutionObject) {
+    // write solution to current problem directory
+    String problemDirectoryPath = null;
+    try {
+      problemDirectoryPath = resourceProvider
+          .getProblemDirectory(getProblemType(), solutionObject.getId())
+          .getAbsolutePath();
+    } catch (IOException e) {
+      solutionObject.setDebugData("Failed to retrieve problem directory.");
+      solutionObject.abort();
+    }
+    return problemDirectoryPath;
+  }
+
+  @Override
+  public Mono<Solution<String>> solve(
+      String input,
+      SubRoutineResolver resolver
+  ) {
+    var solution = new Solution<String>();
+
+    // change "TYPE" keyword from "TSP" to "CVRP"
+    // add capacity declaration of "0" (is ignored later)
+    // this is theoretically wrong, but needed for Lucas' QUBO converter to work
+    if (input.contains("TYPE : TSP")) {
```

Comment:
We only need this in the QuboTspSolver and LkhTspSolver and both instances need to be handeld slighty different. I refactored it so that those are only two lines of code in both instances, i dont want to introduce a Utility class for such small things. (Would have only one line of useful code)

Reply:
Great, that's even better!